packages/flutter/lib/src/widgets/heroes.dart

```diff
@@ -467,8 +467,21 @@ class HeroController extends NavigatorObserver {
     }
   }
 
+  // Disable Hero animations while a user gesture is controlling the navigation.
+  bool _questsEnabled = true;
+
+  @override
+  void didStartUserGesture() {
+    _questsEnabled = false;
+  }
+
+  @override
+  void didStopUserGesture() {
+    _questsEnabled = true;
+  }
+
   void _checkForHeroQuest() {
-    if (_from != null && _to != null && _from != _to) {
+    if (_from != null && _to != null && _from != _to && _questsEnabled) {
```

Comment:
fwiw, i think we'll find that what we actually want is to categorise heroes into those that want to participate during one of these gestures and ones that don't. but that can wait.